src/components/Form/Form.tsx

```diff
@@ -25,14 +25,17 @@ export const Form: FC<FormProps> = ({ activity }) => {
   const { loading, form, error, refetch } = useForm(activity)
   const { t } = useTranslation()
   const [evaluateFormRules] = useEvaluateFormRules(activity.object.id)
-  const { onSubmit } = useSubmitForm({ activity })
+  const { onSubmit, isSubmitting } = useSubmitForm({ activity })
   const { branding } = useHostedSession()
 
   const [formProgress, setFormProgress] = useLocalStorage(activity.id, '')
 
   if (loading) {
     return <LoadingPage title={t('activities.form.loading')} />
   }
+  if (isSubmitting) {
+    return <LoadingPage title={t('activities.form.submitting')} />
+  }
```

Comment:
I don't like this approach. Would prefer if we disabled the submit button on the form component itself while isSubmitting is true. I recognise this means you need to add a prop and handle this in the different form UI components but a layout-level loading page is overkill for a activity-level piece of state